pkg/templates/nstemplatetiers/nstemplatetier_generator.go

```diff
@@ -0,0 +1,187 @@
+// - each key is a tier kind (eg: "basic", "advanced", etc.)
+// - each value is a map of revisions indexed by their associated template kind
+//   (eg: {"code":"123456", "dev":"abcdef", "stage":"cafe01"})
+func parseAllRevisions(metadata []byte) (map[string]map[string]string, error) {
+	data := make(map[string]interface{})
+	err := yaml.Unmarshal([]byte(metadata), &data)
+	if err != nil {
+		return nil, err
+	}
+	// reads all entries in the YAML, splitting the keys
+	// as we expect the following format: '<tier_kind>-<namespace_kind>' (eg: 'advanced-code')
+	revisions := make(map[string]map[string]string)
+	for filename, revision := range data {
+		data := strings.Split(filename, "-")
+		if len(data) != 2 {
+			log.Info("invalid namespace template filename. Expected format: '<tier_kind>-<namespace_kind>'", "filename", filename)
+			continue
+		}
+		tierKind := data[0]
+		nsKind := data[1]
+		// create a new entry if needed
+		if _, ok := revisions[tierKind]; !ok {
+			revisions[tierKind] = make(map[string]string, 3) // expect 3 entries: 'code', 'dev' and 'stage'
+		}
+		revision, ok := revision.(string)
+		if !ok {
+			log.Info("invalid namespace template filename revision. Expected a string", "filename", filename, "revision", revision)
+			continue
+		}
+		revisions[tierKind][nsKind] = revision
+	}
+	log.Info("templates revisions loaded", "revisions", revisions)
+	return revisions, nil
+}
+
+// GenerateManifest generates a full NSTemplateTier manifest
+// by embedding the `<tier>-code.yml`, `<tier>-dev.yml` and `<tier>-stage.yml`
+// file along with each one's git (short) commit as the revision associated with
+// the template.
+//
+// Something like:
+// ------
+// kind: NSTemplateTier
+//   metadata:
+//     name: basic
+//   spec:
+//     namespaces:
+//     - type: code
+//       revision: y8f907f6
+//       template: >
+//         <yaml-ns-template>
+//     - type: dev
+//       revision: f8q907f4
+//       template: >
+//         <yaml-ns-template>
+//     - type: stage
+//       revision: 907fy8f6
+//       template: >
+//         <yaml-ns-template>
+// ------
+func (g NSTemplateTierGenerator) GenerateManifest(tier string) ([]byte, error) {
+	if nstemplatetierTmpl == nil {
+		return nil, errors.Wrapf(initErr, "Cannot generate the '%s' NSTemplateTier manifest", tier)
+	}
+	revisions, ok := g.revisions[tier]
+	if !ok {
+		return nil, errors.Errorf("tier '%s' does not exist", tier)
+	}
+	type namespace struct {
+		Type     string
+		Revision string
+		Template string
+	}
+	namespaces := make([]namespace, 0, len(revisions))
+	// retrieve the namespace types and order them, so we can order them, and compare
+	// with the expected templates during the tests
+	nsTypes := make([]string, 0, len(revisions))
+	for nsType := range revisions {
+		nsTypes = append(nsTypes, nsType)
+	}
+	sort.Strings(nsTypes)
+	for _, nsType := range nsTypes {
+		// get the content of the `-<t>.yaml` file
+		tmpl, err := Asset(fmt.Sprintf("%s-%s.yaml", tier, nsType))
```

Comment:
@alexeykazakov good catch! yes, I already used the `asset` on line 46, but I missed that one since for now both `deploy/` and `test/` pkgs have the same namespace templates.

Reply:
fixed in 74d08a5